docs/documentation/api-specification/mi-api.md

```diff
@@ -27,7 +27,136 @@ Checklist for requesting approval for your extensions.
 
 ---
 ## Description
+MI API allows to write and overwrite parameters sent into transactions. Supports both single and multiple transactions, custom logic and validation.  Can be used to abort transaction based on own validation. It is not possible to add extra fields to transactions. Provides error managment. 
 
 ## Features
 
+### Getting value from data container
+mi.in containes for In values for Transactions. Contained values are already converted to the proper type.
+Takes string key as a parameter and returns value associeted with the key.
+
+Example 1:
+```groovy
+String raw = mi.in.get(name)
+```
+<br>
+
+Example 2:
+```groovy
+  private String whlo;
+  private String itno;
+  private String whsl;
+
+  public void main() {
+    whlo = mi.inData.get("WHLO").trim();
+    itno = mi.inData.get("ITNO").trim();
+    whsl = mi.inData.get("WHSL").trim();
+
+  if (!validateInput()) return;
+```
+
+### Putting value into data container
+Takes key and associeted with it value as parameters, returns previous value.
+Example 1:
+```groovy
+ public void main() {
+    mi.outData.put("WHLO", "This is WHLO Output#1 for QQS001")
+    mi.outData.put("ITNO", "This is ITNO Output#2 for QQS002")
+    mi.write() 
+  }
+```
+<br>
+
+Example 2:
+```groovy
+void main() {
+    mi.outData.put("WHLO", String.valueOf(getParameter("NUMB", Double.class).orElse(null)))
+  }
+```
+### Checking date format
+Returns date format as a string. Possible formats: YMD8, YMD6, MDY6, DMY6, YWD5
+Example:
+```groovy
+void main() {
+    return mi.getDateFormat(date1)
+  }
+```
+### Checking date separator
+Returns date separator as a char.
+Example:
+```groovy
+void ReadSeparator {
+sep = mi.getDateSeparator() 
+String oldDate = '04-DEC-2012'
+Date date = Date.parse( 'dd-MMM-yyyy', oldDate )
+String newDate = date.format( 'M'{sep}'d'{sep}'yyyy' )
+return newDate
+}
+```
+
+### Checking remaining records
+Returns the amount of remaining records allowed to be written in current transaction.
+Example:
+```groovy
+void main() {
+    if mi.getRemainingRecords(whlo)>=2{
+      whsl = mi.inData.get("WHSL")
+    }
+  }
+```
+
+### Chcecking if there are remaining records
+Returns true if there are still any additional records to be written or false is not
+Example:
+```groovy
+void main() {
+    if mi.hasRemainingRecords(whlo) == true
+      whsl = mi.inData.get("WHSL")
+    }
+  }
+```
+
+### Chcecking maximum of possible records
+Returns the maximum amount of possible records to be written as an intiger.
+Example:
+```groovy
+void main() {
+    return mi.getMaxRecords(whlo)
```

Comment:
mi.getMaxRecords cannot be called this way, we can discuss the use case separately